lambdas/ecs_ec2_instance_tagger/ecs_ec2_instance_tagger.py

```diff
@@ -46,8 +59,35 @@ def main(event, _):
     print(f'event = {event!r}')
 
     ec2_client = boto3.client('ec2')
+    s3_client = boto3.client('s3')
+
+    bucket_name = os.environ["BUCKET_NAME"]
+    object_path = os.environ["OBJECT_PATH"]
 
     ec2_instance_id = event['detail']['ec2InstanceId']
 
-    response = create_tags(ec2_client, ec2_instance_id, event['detail'])
-    print(f'response = {response!r}')
+    try:
+        s3_client.get_object(
+            Bucket=bucket_name,
+            Key=f'{object_path}/{ec2_instance_id}'
+        )
+    except ClientError as ex:
+        if ex.response['Error']['Code'] == 'NoSuchKey':
+            print(f'{ec2_instance_id} not seen yet, tagging!')
+
+            response = create_tags(
+                ec2_client,
+                ec2_instance_id,
+                event['detail'])
+
+            print(f'response = {response!r}')
+        else:
+            raise ex
```

Comment:
minor: if you’re in the body of an `except:` in Python, you can just call `raise` and it automatically re-raises the exception.